src/lib/cmd.rs

```diff
@@ -1,64 +1,101 @@
 use std::fs::File;
 use std::process::{Command, Stdio};
+use std::sync::Arc;
+use tokio::sync::RwLock;
+use uuid::Uuid;
+use nix::sys::signal::{self, Signal};
+
+use crate::lib::job;
 
 use super::config;
 
-fn remote_cmd(args: &mut Vec<&str>, output: &mut File) -> Result<(), String> {
-    // TODO(azhng): let's finger cross this works.
+async fn remote_cmd(job_id: &Uuid, args: &mut Vec<&str>, output: &mut File, job_registry: &Arc<RwLock<job::JobRegistry>>) -> Result<(), String> {
     let output_file = output.try_clone().unwrap();
     let error_file = output.try_clone().unwrap();
-    let cmd = Command::new("ssh")
+    let cmd_process = Command::new("ssh")
         .arg(config::SEXXI_REMOTE_HOST)
-        .args(args)
+        .args(&*args)
         .stdout(Stdio::from(output_file))
         .stderr(Stdio::from(error_file))
-        .output()
-        .expect("Ok");
+        .spawn();
+
+    match cmd_process {
+        Ok(child) => {
+            info!("Server process with pid {} is running command `{:?}`", child.id(), args);
+            {
+                let job_registry = job_registry.read().await;
+                let job: Arc<RwLock<job::JobDesc>>;
+                if let Some(j) = job_registry.jobs.get(job_id) {
+                    job = j.clone();
+                    let mut job = job.write().await;
+                    job.pid = child.id();
+                }
+            }
+
+            let result = child.wait_with_output().expect("Ok");
+            if !result.status.success() {
+                /*
+                   Check if Signal::SIGKILL was received. If it was the case
+                   this is probably because the the job was canceled.
+                */
+                if result.status.to_string() == "signal: 9" {
+                    let job_registry = job_registry.read().await;
+                    let job: Arc<RwLock<job::JobDesc>>;
+                    if let Some(j) = job_registry.jobs.get(job_id) {
+                        job = j.clone();
+                        let mut job = job.write().await;
+                        job.status = job::JobStatus::Canceled;
```

Comment:
Probably want to set github status here accordingly.

Reply:
So from Github, there are no status for cancelled job, so I think this should not been done:

- `failure` if any of the contexts report as `error` or `failure`
- `pending` if there are no statuses or a context is `pending`
- `success` if the latest status for all contexts is `success`

Reply:
Make an issue to investigate. This is not related to the issue I am working on which is simply tracking the PID

Reply:
we should remove pending then, if we are setting the status pending

Reply:
if not then all good